fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt

```diff
@@ -69,8 +69,20 @@ class Fretboard(
      * @param block block of code to be executed if the user is part of the experiment
      */
     fun withExperiment(context: Context, descriptor: ExperimentDescriptor, block: (Experiment) -> Unit) {
-        if (evaluator.evaluate(context, descriptor, experiments)) {
-            block(experiments.first { it.id == descriptor.id })
+        val experiment = evaluator.evaluate(context, descriptor, experiments)
+        if (experiment != null) {
+            block(experiment)
         }
     }
+
+    /**
+     * Gets the metadata associated with the specified experiment
```

Comment:
Let's mention explicitly that it will return the experiment even if the user is not part of it!

Reply:
Yes, that's an important point, I'm going to add it